module/zfs/dbuf.c

```diff
@@ -1764,6 +1764,96 @@ dbuf_unoverride(dbuf_dirty_record_t *dr)
 }
 
 /*
+ * Disassociate the frontend for any older transaction groups of a
+ * dbuf that is inside a range being freed.  The primary purpose is to
+ * ensure that the state of any dirty records affected by the operation
+ * remain consistent.
+ */
+static void
+dbuf_free_range_disassociate_frontend(dmu_buf_impl_t *db, dnode_t *dn,
```

Comment:
I'm not sure what exactly "frontend" means here.  How about `dbuf_free_range_disassociate_dirty()` instead.